.github/workflows/sm2.yml

```diff
@@ -19,44 +19,22 @@ env:
   RUSTDOCFLAGS: "-Dwarnings"
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        rust:
-          - 1.81.0 # MSRV
-          - stable
-        target:
-          - thumbv7em-none-eabi
-          - wasm32-unknown-unknown
-    steps:
-      - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: ${{ matrix.rust }}
-          targets: ${{ matrix.target }}
-      - run: cargo build --target ${{ matrix.target }} --release --no-default-features
-      - run: cargo build --target ${{ matrix.target }} --release --no-default-features --features alloc
-      - run: cargo build --target ${{ matrix.target }} --release --no-default-features --features arithmetic
-      - run: cargo build --target ${{ matrix.target }} --release --no-default-features --features pkcs8
-      - run: cargo build --target ${{ matrix.target }} --release --no-default-features --features alloc,arithmetic,pkcs8
```

Comment:
Why did you remove this? It's how we check for `no_std` compatibility (perhaps it should be renamed to `no_std`)

Reply:
getrandom now requires the backend to be specified. It won't work with either thumbv7 nor wasm32 without the consumer of the crate specifying it.

For wasm32 I could make a mock consumer-crate for this purpose and inject the `wasm_js` feature flag, but for thumbv7 I don't have a getrandom backend in the first place.
I can't pull `wasm_js` either directly.

Reply:
Actually? `getrandom` shouldn't be pulled in the first place? There is something going on in that dependency tree.

Reply:
That last commit should fix that. (now requires an external Rng)